jax/numpy/lax_numpy.py

```diff
@@ -3150,6 +3150,21 @@ def sort(a, axis=-1, kind='quicksort', order=None):
   else:
     return lax.sort(a, dimension=_canonicalize_axis(axis, ndim(a)))
 
+@_wraps(np.sort_complex)
+def sort_complex(a):
+    a = lax.sort(a, dimension = 0)
+    if not issubclass(a.dtype.type, complexfloating):
```

Comment:
Couldn't we write this more simply something like this:
```
a = lax.sort(a, dimension=0)
return lax.convert_element_type(a, result_type(a, complex64))
```

`lax.convert_element_type` is a no-op if the type conversion is redundant, and `jnp.result_type` has the "promote to complex" behavior you are looking for.
?